### PR TITLE
Refresh entities after resubmit

### PR DESCRIPTION
**Pull request by ErikMogensen**
This PR makes the affected entities in the tree view refresh after doing a repair and resubmit.

### CODE REVIEW

**Comment by SeanFeldman** [comment]
:white_check_mark: [Build ServiceBusExplorer 5.1.0-RefreshEntityAfterResubmit.24.build.787 completed](https://ci.appveyor.com/project/seanfeldman/servicebusexplorer/builds/37306786) (commit https://github.com/paolosalvatori/ServiceBusExplorer/commit/7726e5b88e by @ErikMogensen)

**Comment by SeanFeldman** [comment]
:white_check_mark: [Build ServiceBusExplorer 5.1.0-RefreshEntityAfterResubmit.25.build.788 completed](https://ci.appveyor.com/project/seanfeldman/servicebusexplorer/builds/37306809) (commit https://github.com/paolosalvatori/ServiceBusExplorer/commit/efadae5005 by @ErikMogensen)

**Comment by ErikMogensen** [comment]
Apparently GiHub has problems showing the diff in MainForm.cs. It looks like I changed almost everything in it, which I didn't.

File: src/ServiceBusExplorer/Forms/MainForm.cs [hunk 1]
```diff
@@ -1698,1000 +1695,628 @@ private void renameEntity_Click(object sender, EventArgs e)
...
+                                    await serviceBusHelper.DeleteRelays(relayServiceList);
+                                }
+                            }
+                        }
                         return;
                     }
-                    // Relays Node (Create New RelayDescription)
-                    if (serviceBusTreeView.SelectedNode.Text == Constants.RelayEntities)
+                    // Queue Node
+                    if (serviceBusTreeView.SelectedNode.Tag is QueueDescription)
```
Comment (SeanFeldman) [inline src/ServiceBusExplorer/Forms/MainForm.cs:2121]: Nit: Type pattern could be used for scenarios such as this one to remove repeated code below.

**Review by SeanFeldman — APPROVED** [review]
Optional: see the comment below. Since it can be applied in multiple places in MainForm alone, could be just a stand-alone PR.

**Comment by ErikMogensen** [comment]
> Nit: Type pattern could be used for scenarios such as this one to remove repeated code below.

Great idea. I'll also turn some of those if statements into switch blocks in this PR. I assume that the reason for it having that many if's is that SBE started as a tool for one service then others came along, one by one.

**Comment by SeanFeldman** [comment]
:white_check_mark: [Build ServiceBusExplorer 5.1.0-RefreshEntityAfterResubmit.27.build.789 completed](https://ci.appveyor.com/project/seanfeldman/servicebusexplorer/builds/37384600) (commit https://github.com/paolosalvatori/ServiceBusExplorer/commit/e5d373c2c2 by @ErikMogensen)

**Comment by ErikMogensen** [comment]
Used the type matching pattern to simply some parts. Also simplified some tests using the null conditional operator.

I did not turn any if's into switch statements due to the effort it would require so I leave that for another PR. @paolosalvatori , I assume you wrote most of MainForm event handling code, do you think it is a good idea turning some of those if statements into switch/case?

**Comment by paolosalvatori** [comment]
@ErikMogensen when I started to write the tool there was just Service Bus, then other services were added over time as you know. This is why there are "if" statements and not switch in the code of the MainForm. Yes, you can turn "if" statements into "switch" where it makes sense :)

**Review by paolosalvatori — APPROVED** [review]

**Comment by SeanFeldman** [comment]
:white_check_mark: [Build ServiceBusExplorer 5.1.0-RefreshEntityAfterResubmit.31.build.797 completed](https://ci.appveyor.com/project/seanfeldman/servicebusexplorer/builds/37514079) (commit https://github.com/paolosalvatori/ServiceBusExplorer/commit/9575b47152 by @ErikMogensen)

**Comment by ErikMogensen** [comment]
That was my initial thought @paolosalvatori but when I started looking into it I saw that many if statements were comparing against variables and until C# 6 or 7 you had to use constants for comparison so limitations of the language were also blocking at the time you wrote it.

I won't do anything else to this PR unless you or @SeanFeldman find something that needs to be fixed. It requires both of you to approve it again since I had to make a change for a merge conflict. I am thinking of switching to the latest SDKs for the part where it populates the tree view. In that case I would probably change it to switch statements to make it more readable.

**Review by SeanFeldman — APPROVED** [review]

**Review by paolosalvatori — APPROVED** [review]
